solr/solr-ref-guide/modules/deployment-guide/pages/distributed-tracing.adoc

```diff
@@ -25,6 +25,17 @@ A sampled distributed tracing query request displayed in Jaeger UI looks like th
 .Tracing of a Solr query
 image::distributed-tracing/query-request-tracing.png[]
 
+== Always-on Trace Id generation
+
+Starting from 9.4 Solr ships with an always-on trace id generator. This will inject trace id headers for every Solr request, propagating the client supplied value or generating a new id as needed and replaces the existing `rid` mechanism.
+
+The header name it uses for propagation is `X-Trace-Id`, this can be changed by updating the system property `solr.traceIdHeader`.
+
+This plugin will not be enabled if another Tracing mechanism is setup via Solr config.
+This plugin can be disabled by setting the `solr.alwaysOnTraceId` system property to `false`.
+
+Note: Solr 9.4 ships with an `OpenTracing` based lightweight `Tracer` implementation and Solr 10 has an updated `OpenTelemetry` based `TextMapPropagator` implementation.
+
```

Comment:
@ctargett at some point did you say something about reference guide style in which we shouldn't refer to older or newer versions?  After all, the ref guide is itself versioned.  I don't think you said exactly that but I recall you said something similar.

Reply:
Yes, you got the gist of what I meant correctly.

One of the challenges of including notes about behavior in specific versions is that at some point it becomes obsolete. A reader of the 9.4 Ref Guide may find it important to know that this feature is added in 9.4, but does the reader of 9.6? 10.0? One needs to remove such comments at some point, which turns it into a maintenance challenge.

Additionally, what is the value of telling someone in 9.4 that Solr 10 is going to include a different implementation? This comment simply mentions it in passing - does it impact how they implement in 9.4? If not, it should be left for the 10 release notes.

Reply:
paragraph updated. please let me know if it looks better